### PR TITLE
Adding LendFlare finance on ethereum chain to defillama adapters .

### PR DESCRIPTION
**Pull request by LendFlare**
Twitter Link: https://twitter.com/LendFlareOffic
List of audit links if any: https://www.certik.com/projects/lend-flare
Website Link: https://lendflare.finance
Logo (High resolution, preferably in .svg and .png, for application on both white and black backgrounds. Will be shown with rounded borders): 
LendFlare-Black
LendFlare-White
LendFlare_Black
LendFlare_White
Current TVL: $36,702,471.80
Chain: Ethereum
Coingecko ID (so your TVL can appear on Coingecko): lend-flare-dao-token
Coinmarketcap ID (so your TVL can appear on Coinmarketcap): 19189
Short Description (to be shown on DefiLlama): Short Description (to be shown on DefiLlama): Lendflare is a decentralized borrowing platform on Ethereum blockchain that allows Curve LP holders (the borrowers) to draw fixed-rate, fixed term and high LTV loans against Curve LP used as collateral, with no concerns for assets being liquidated due to price fluctuation. Loans are paid out through Lendflare from Compound platform. Liquidity providers (the lenders) who deposit loan liquidity through Lendflare on Compound platform will receive one of the highest interest rate compared to other current lending platforms.
Token address and ticker if any: https://etherscan.io/token/0xB620Be8a1949AA9532e6a3510132864EF9Bc3F82
Category (full list at https://defillama.com/categories) *Please choose only one: https://defillama.com/protocols/Lending
Oracle used (Chainlink/Band/API3/TWAP or any other that you are using): No
forkedFrom (Does your project originate from another project): No
methodology (what is being counted as tvl, how is tvl being calculated): LendFlare was launched on 8th of March 2022, until 7th of April 2022, count all supply and borrow TVL.

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/lendflare/index.js exports TVL: 
        
 
 
    ethereum                  37.30 M
    
    total                    37.30 M

**Comment by g1nt0ki** [comment]
Hi, 

Thanks for the PR. It looks really good, but one change is needed, can you pass the `block` value to different methods and include it in sdk `call` and `multicall` method, this way we want to compute tvl at different blocks/days

File: projects/lendflare/uniswap.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+const sdk = require('@defillama/sdk');
```
Comment (g1nt0ki) [inline projects/lendflare/uniswap.js:1]: this module is not used right, can it be removed?

File: projects/lendflare/index.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+const sdk = require('@defillama/sdk');
+const convexBooster = require('./convexBooster.js');
+const supplyBooster = require('./supplyBooster.js');
+const uniswap = require('./uniswap.js');
+const { toUSDTBalances } = require('./utils.js');
+
+async function tvl(timestamp, block, chainBlocks) {
```
Comment (g1nt0ki) [inline projects/lendflare/index.js:7]: this is the `block` variable that I was talking about, can you forward to tvl calls and use it

**Comment by LendFlare** [comment]
@g1nt0ki OK, I'll adjust it.

**Comment by LendFlare** [comment]
Thank you for your guidance. We will provide new statistics and readjust the display mode in the future. For now we want to use this code first.The new code has been submitted and hopes to be merged.


Thx @g1nt0ki

**Comment by llamatester** [comment]
Error while running adapter at projects/lendflare/index.js: 
        
 
  
    
    
    ReferenceError: chainBlocks is not defined
        at getTotalSupply (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/lendflare/supplyBooster.js:18:16)
        at Object.tvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/lendflare/supplyBooster.js:41:25)
        at tvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/lendflare/index.js:7:40)
        at processTicksAndRejections (node:internal/process/task_queues:96:5)
        at async getTvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:57:25)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:149:9
        at async Promise.all (index 0)
        at async Promise.all (index 1)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:194:3

File: projects/lendflare/supplyBooster.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    { virtualBalance: "0x10A377BA353cd480E43800018a27742054904c1a", decimals: 6, coinName: "usd-coin" },
+    // dai
+    { virtualBalance: "0x2FbE41e4861B5d0936AA01bB32bd3402d34f11fA", decimals: 18, coinName: "dai" },
+    // wbtc
+    { virtualBalance: "0xcc29655C9F9A211fb11c25D905306aa93a685ef3", decimals: 8, coinName: "wrapped-bitcoin" },
+    // eth
+    { virtualBalance: "0x6d18E830A938F0eAF206f1BD80b79a851E5f37A3", decimals: 18, coinName: "ethereum" },
+]
+
+async function getTotalSupply(pools) {
```
Comment (g1nt0ki) [inline projects/lendflare/supplyBooster.js:16]: this function is missing a few parameters

File: projects/lendflare/convexBooster.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    return pools;
+}
+
+async function tvl(timestamp, block, chainBlocks) {
+    let tvl = new BN(0);
+
+    const pools = await getTotalSupply(convexPools, timestamp, block, chainBlocks).then(pools => {
+        return calculateTokenAmount(pools, timestamp, block, chainBlocks);
+    });
+    const prices = await utils.getPricesfromString().then(result => {
```
Comment (g1nt0ki) [inline projects/lendflare/convexBooster.js:110]: sorry, missed this part. You probably need to pass the block data here as well, so you fetch the right prices

File: projects/lendflare/utils.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+            "internalType": "uint256",
+            "name": "",
+            "type": "uint256"
+        }
+    ],
+    "stateMutability": "view",
+    "type": "function"
+}
+
+// ethereum
```
Comment (g1nt0ki) [inline projects/lendflare/utils.js:22]: repeat of the comment, the block needs to be passed to the methods in this file as well

**Comment by g1nt0ki** [comment]
you can locally verify if this adapter works by running `node test.js projects/lendflare/index.js `

**Comment by llamatester** [comment]
The adapter at projects/lendflare/index.js exports TVL: 
        
 
 
    ethereum                  39.21 M
    
    total                    39.21 M

**Comment by g1nt0ki** [comment]
@LendFlare your list should appear in Defillama in hopefully under an hour

**Pull request merged**